core/src/main/java/com/uber/m3/tally/AbstractBuckets.java

```diff
@@ -21,30 +21,39 @@
 package com.uber.m3.tally;
 
 import com.uber.m3.util.Duration;
+import com.uber.m3.util.ImmutableList;
 
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
 import java.util.ListIterator;
 
 /**
- * Abstract {@link Buckets} implementation for common functionality.
+ * @deprecated DO NOT USE
+ *
+ * Please use {@link ImmutableBuckets} instead
  */
-public abstract class AbstractBuckets<T> implements Buckets<T> {
+@Deprecated
+public abstract class AbstractBuckets<T extends Comparable<T>> implements Buckets<T> {
     protected List<T> buckets;
 
     AbstractBuckets(T[] buckets) {
         if (buckets == null) {
-            this.buckets = new ArrayList<>();
-        } else {
-            this.buckets = new ArrayList<>(Arrays.asList(buckets));
+            throw new IllegalArgumentException("provided buckets could not be null");
```

Comment:
confirming that this is is now ok because the Scope always provides non-null buckets because of #76, yeah?

Reply:
Yes